### PR TITLE
riscv: device tree + virt board + uart8250 + fix

### PR DESCRIPTION
**Pull request by Supercip971**

### CODE REVIEW

File: sources/libs/hw/fdt/fdt.h [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+#define FDT_MAGIC 0xd00dfeed
+
+FdtBeginNode *fdt_root(FdtHeader *fdt);
+
+void dump_fdt(Emit *emit, FdtHeader *header, FdtBeginNode *current_fdt);
+
+FdtBeginNode *get_fdt_node(FdtBeginNode *node, Str entry);
+
+NodeProperty get_fdt_property(FdtHeader *header, FdtBeginNode *node, Str name);
```
Comment (sleepy-monax) [inline sources/libs/hw/fdt/fdt.h:65]: We are not getting, we are doing a lookup.
Also, we should avoid get/set (see https://www.youtube.com/watch?v=ZsHMHukIlJY&t=2351s)

Also please namespace identifiers like:

fdt_root
fdt_dump
fdt_lookup_node
fdt_lookup_property

File: sources/libs/hw/fdt/fdt.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+typedef enum
+{
+    FDT_BEGIN_NODE = 1,
+    FDT_END_NODE = 2,
+    FDT_PROP = 3,
+    FDT_NOP = 4,
+    FDT_END = 9
+} FdtNodeType;
+
+typedef struct PACKED ALIGNED(4)
```
Comment (sleepy-monax) [inline sources/libs/hw/fdt/fdt.h:36]: Is ALIGNED(4) really needed?

File: sources/libs/hw/fdt/fdt.h [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    be_uint32_t name_offset;
+    char value[];
+} FdtPropertiesNode;
+
+typedef struct
+{
+    Str property_name;
+    size_t property_value_size;
+    void *property_value;
+} NodeProperty;
```
Comment (sleepy-monax) [inline sources/libs/hw/fdt/fdt.h:55]: The property_* prefix is not useful, we already know it's a property from the type name 

Also, it's missing the Fdt prefix

File: sources/libs/hw/fdt/fdt.h (hunk 2 above)
Comment (Supercip971) [reply sources/libs/hw/fdt/fdt.h:36]: yes, gcc may spit warning

Comment (sleepy-monax) [reply sources/libs/hw/fdt/fdt.h:36]: What warning?

Comment (Supercip971) [reply sources/libs/hw/fdt/fdt.h:36]: see: https://github.com/brutal-org/brutal/runs/4304992045?check_suite_focus=true

Comment (sleepy-monax) [reply sources/libs/hw/fdt/fdt.h:36]: Why are we casting to `be_uint32_t *`?

Comment (Supercip971) [reply sources/libs/hw/fdt/fdt.h:36]: because we are treating type as a big endian

Comment (sleepy-monax) [reply sources/libs/hw/fdt/fdt.h:36]: We already have a type field in the struct which is big endian

**Review by sleepy-monax — APPROVED** [review]

**Pull request merged**